### DISCUSSION TITLE
`context.session.$authorize()` vs `resolver.authorize()`

### DISCUSSION THREAD

**Discussion by antonykamp** [blitz-js/blitz#2472] [Q&A] [answered]
Hey, 
I thought, that `context.session.$authorize()` and `resolver.authorize()` are both verifying and validating the users' role and login-status. But when I try the following, the `user` is either a number or null (the login-status isn't validated) :/

```js
import { Ctx, resolver } from "blitz"
export default resolver.pipe(resolver.authorize(), async ({ id }, context: Ctx) => {
  const user = context.session.userId  // user: number | null
})
```

When I use `context.session.$authorize()` instead `user` is a number (the login-status is validated).

```js
import { Ctx, resolver } from "blitz"
export default resolver.pipe(async ({ id }, context: Ctx) => {
  context.session.$authorize()
  const user = context.session.userId  // user: number
})
```

I thought `resolver.authorize()` is based on `context.session.$authorize()`? But it doesn't validate the input like `resolver.zod` does. Am I doing something wrong?

antony :)

**Comment by flybayer** [comment, marked as answer]
Here's the fix. Don't use `Ctx` when you use `resolver.x` :)

```diff
-import { Ctx, resolver } from "blitz"
+import { resolver } from "blitz"
-export default resolver.pipe(resolver.authorize(), async ({ id }, context: Ctx) => {
+export default resolver.pipe(resolver.authorize(), async ({ id }, context) => {
  const user = context.session.userId  // user: number | null
})
```

**Reply by antonykamp** [reply]
Thank you so much 🙏

**Reply by oyatek** [reply]
why so and what's the type of `context ` then? Should it be `BlitzCtx`?
